### PR TITLE
Fix the last of the coverity defects. 

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: common/rpc/RpcChannel.cpp [hunk 1]
```diff
@@ -63,12 +63,18 @@ class OutstandingRequest {
    * These are requests on the server end that haven't completed yet.
    */
  public:
-    OutstandingRequest() {}
-    ~OutstandingRequest() {}
-
-    int id;
-    RpcController *controller;
-    google::protobuf::Message *response;
+  OutstandingRequest(int id,
```
Comment (peternewman) [inline common/rpc/RpcChannel.cpp]: Didn't you change this once already? :)

File: common/thread/ConsumerThread.cpp [hunk 2]
```diff
@@ -64,6 +64,7 @@ void ConsumerThread::EmptyQueue() {
     action->Run();
 
     // reacquire the lock
+    // coverity[LOCK]
```
Comment (peternewman) [inline common/thread/ConsumerThread.cpp]: Can you add a bit more detail in the comment above why we finish with the lock?

File: common/rpc/RpcChannel.cpp (hunk 1 above)
Comment (nomis52) [inline common/rpc/RpcChannel.cpp]: That was the OutstandingResponse, I missed the Request last time

File: doxygen/rpc.dox [hunk 3]
```diff
@@ -87,7 +87,8 @@
  *
  * We can generate the C++ code by running:
  *
- *     protoc/ola_protoc --cppservice_out ./ Dictionary.proto
+ *     protoc --plugin=protoc-gen-cppservice=protoc/ola_protoc_plugin" \
```
Comment (peternewman) [inline doxygen/rpc.dox]: Unmatched quote. Also what's breaking the windows build is the lack of EXEEXT, but I guess that's taken for granted in a code example?

File: common/thread/ConsumerThread.cpp (hunk 2 above)
Comment (nomis52) [inline common/thread/ConsumerThread.cpp]: It's in the @pre and @post conditions in the comments.

File: doxygen/rpc.dox [hunk 4]
```diff
@@ -87,7 +87,8 @@
...
+ *     --cppservice_out ./ Dictionary.proto
```
Comment (peternewman) [inline doxygen/rpc.dox]: Shouldn't there be a --proto-path before Dictionary.proto?

Comment (peternewman) [inline doxygen/rpc.dox]: I'm guessing we can't test this like the doxygen examples?

File: plugins/karate/KarateLight.cpp [hunk 5]
```diff
@@ -253,6 +253,12 @@ bool KarateLight::ReadBack(uint8_t *rd_data, uint8_t *rd_len) {
 
   // read payload-data (if there is any)
   uint8_t payload_size = rd_buffer[CMD_HD_LEN];
+  if (payload_size > CMD_MAX_LENGTH - CMD_HD_LEN) {
```
Comment (peternewman) [inline plugins/karate/KarateLight.cpp]: Round brackets please.

**Comment by peternewman** [comment]
LGTM

File: common/rpc/RpcChannel.cpp (hunk 1 above)
Comment (peternewman) [inline common/rpc/RpcChannel.cpp]: Glad I'm not going crazy!

File: common/thread/ConsumerThread.cpp (hunk 2 above)
Comment (peternewman) [inline common/thread/ConsumerThread.cpp]: Ah yes, hidden in the code review.

File: doxygen/rpc.dox (hunk 4 above)
Comment (nomis52) [inline doxygen/rpc.dox]: I've confirmed this works, and yeah, since it's documentation we can't test it

File: doxygen/rpc.dox (hunk 3 above)
Comment (nomis52) [inline doxygen/rpc.dox]: Fixed.

File: plugins/karate/KarateLight.cpp (hunk 5 above)
Comment (nomis52) [inline plugins/karate/KarateLight.cpp]: Not here - they only add noise. 

We really have to stop you add parens all over the place :)

Comment (peternewman) [inline plugins/karate/KarateLight.cpp]: Never :)

**Comment by peternewman** [comment]
LGTM

**Pull request merged**